user-interface/src/main/java/life/qbic/datamanager/views/projects/project/measurements/MeasurementPresenter.java

```diff
@@ -42,16 +44,16 @@ private static ProteomicsMeasurementEntry convertProteomicsMeasurement(
   }
 
   private static NGSMeasurementEntry convertNGSMeasurement(NGSMeasurement measurement,
-      SampleInformation sampleInfo) {
+      SampleInformation sampleInfo, NGSSpecificMeasurementMetadata specificMeasurementMetadata) {
     return new NGSMeasurementEntry(measurement.measurementCode().value(),
         sampleInfo, measurement.organisation().IRI(), measurement.organisation().label(),
         measurement.instrument().getName().replace("_", ":"),
         measurement.instrument().getLabel(),
         measurement.samplePoolGroup().orElse(""), measurement.facility(),
         measurement.sequencingReadType(),
         measurement.libraryKit().orElse(""), measurement.flowCell().orElse(""),
-        measurement.sequencingRunProtocol().orElse(""), measurement.indexI7().orElse(""),
-        measurement.indexI5().orElse(""), measurement.comment().orElse(""));
+        measurement.sequencingRunProtocol().orElse(""), specificMeasurementMetadata.index().orElse(new NGSIndex("", "")).indexI7(),
+        specificMeasurementMetadata.index().orElse(new NGSIndex("", "")).indexI5(), specificMeasurementMetadata.comment().orElse(""));
```

Comment:
Optional#map to the rescue, no need to create an empty index here.

Reply:
I just noticed that this might not be the case as empty index neither is an Optional nor null. I initially read the `isEmpty()` as the `isEmpty()` method of the `Optional` class. It turned out it is the `isEmpty()` method of the `String` class.